tests/algorithms/test_dagger.py

```diff
@@ -325,6 +366,69 @@ def test_trainer_makes_progress(init_trainer_fn, pendulum_venv, pendulum_expert_
     )
 
 
+@pytest.mark.parametrize(
+    "init_trainer_fn",
+    [_build_dagger_trainer, _build_simple_dagger_trainer],
+)
+def test_trainer_reproducible(
+    init_trainer_fn,
+    tmpdir,
+    pendulum_venv,
+    pendulum_expert_policy,
+    custom_logger,
+):
+    # Check that we get the same results if we run the trainer
+    # twice with the same random seeds.
+    # In particular, check that the rewards achieved by the imitation
+    # policy are the same in each run.
+    rewards_list = []
+    with torch.random.fork_rng():
+        for run_idx in range(2):
+            # Reset the random seeds.
+            run_dir = os.path.join(tmpdir, "run{0}".format(run_idx))
+            torch.random.manual_seed(12345)
+            rng = np.random.default_rng(12345)
+            pendulum_venv.seed(12345)
+            pendulum_venv.action_space.seed(12345)
+
+            beta_schedule = None
+            maybe_pendulum_expert_trajectories = None
+            trainer = init_trainer_fn(
+                run_dir,
+                pendulum_venv,
+                beta_schedule,
+                pendulum_expert_policy,
+                maybe_pendulum_expert_trajectories,
+                custom_logger,
+                rng,
+            )
+
+            # Train for 10 iterations. (6 or less causes test to fail on some configs.)
```

Comment:
Is this comment / the 10 iteration minimum still relevant? I think the test you copied it from was checking reward improved by some significant threshold, which needs multiple iterations. But here you're just trying to check the results line up, so can probably get away with fewer iterations.